### PR TITLE
[go-gen] Add returning responses to CREATE and UPDATE endpoints

### PR DESCRIPTION
**Pull request by willjvsmith**
* Removes unnecessary error from `executeQueryWithRowResponse`
* Reorders DAO structs to follow List, then CRUD
* Adds the returning of responses for CREATE and UPDATE endpoints
  * Except in the special case of the Auth service
* Renames gets to reads

I've tested this myself, but I have little confidence that there are no mistakes in these services, hence I have created a card for adding unit tests.

### CODE REVIEW

File: auth/dao/dao.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (dao *DAO) CreateAuth(request AuthCreateRequest) error {
 
 // ReadAuth attempts to find an existing auth'd user in the data store
 func (dao *DAO) ReadAuth(request AuthReadRequest) (*AuthReadRequest, error) {
-	row, err := executeQueryWithRowResponse(dao.DB, "SELECT email, password FROM auth WHERE email = $1", request.Email)
+	row := executeQueryWithRowResponse(dao.DB, "SELECT email, password FROM auth WHERE email = $1", request.Email)
 	var auth AuthReadRequest
 	err = row.Scan(&auth.Email, &auth.Password)
```
Comment (jaylees14) [inline auth/dao/dao.go]: Looks like a compiler error, this should fix it
```suggestion
	err := row.Scan(&auth.Email, &auth.Password)
```

**Review by jaylees14 — APPROVED** [review]
LGTM

File: match/dao/dao.go [hunk 2]
```diff
@@ -71,79 +87,89 @@ func (dao *DAO) Init(config *util.Config) error {
...
 	if err != nil {
 		return nil, err
 	}
 
-	var match MatchGetResponse
-	err = row.Scan(&match.ID, &match.UserOne, &match.UserTwo, &match.MatchedOn)
+	return &resp, nil
+}
+
+// ReaddMatch returns the information about a match stored for a given ID
```
Comment (c-harding) [inline match/dao/dao.go]:
```suggestion
// ReadMatch returns the information about a match stored for a given ID
```

**Review by c-harding — APPROVED** [review]

**Review by Lewiky — APPROVED** [review]
LGTM

**Pull request merged**